### PR TITLE
Fix mounting on QEMU with socket_vmnet

### PR DESCRIPTION
**Pull request by spowelljr**
Closes https://github.com/kubernetes/minikube/issues/15107

**Before:**
```
$ minikube mount ./thing:/mount123
📁  Mounting host path ./thing into VM as /mount123 ...
    ▪ Mount type:   
    ▪ User ID:      docker
    ▪ Group ID:     docker
    ▪ Version:      9p2000.L
    ▪ Message Size: 262144
    ▪ Options:      map[]
    ▪ Bind Address: 192.168.105.6:58140
🚀  Userspace file server: ufs starting
🛑  Userspace file server is shutdown

❌  Exiting due to GUEST_MOUNT: mount with cmd /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=58140,trans=tcp,version=9p2000.L 192.168.105.6 /mount123" : /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=58140,trans=tcp,version=9p2000.L 192.168.105.6 /mount123": Process exited with status 32
stdout:

stderr:
mount: /mount123: mount(2) system call failed: Connection refused.


╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                                                        │
│    😿  If the above advice does not help, please let us know:                                                          │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                                                        │
│                                                                                                                        │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.                                 │
│    Please also attach the following file to the GitHub issue:                                                          │
│    - /var/folders/9l/6wpxv6wd1b901m1146r579wc00rqw3/T/minikube_mount_6b51cf7e154ef7ec9dd71a511f021b37ee49a5f9_0.log    │
│                                                                                                                        │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

**After:**
```
$ minikube mount ./thing:/mount123
📁  Mounting host path ./thing into VM as /mount123 ...
    ▪ Mount type:   
    ▪ User ID:      docker
    ▪ Group ID:     docker
    ▪ Version:      9p2000.L
    ▪ Message Size: 262144
    ▪ Options:      map[]
    ▪ Bind Address: 192.168.105.1:59680
🚀  Userspace file server: ufs starting
✅  Successfully mounted ./thing to /mount123

📌  NOTE: This process must stay alive for the mount to be accessible ...
```

**Trying to mount on `user` network:**
```
$ minikube mount ./thing:/mount123

❌  Exiting due to MK_UNIMPLEMENTED: minikube mount is not currently implemented with the user network on QEMU, try starting minikube with '--network=socket_vmnet'
```

### CODE REVIEW

File: pkg/minikube/cluster/ip.go [hunk 1]
```diff
@@ -68,12 +68,9 @@ func HostIP(host *host.Host, clusterName string) (net.IP, error) {
 		// user network case
 		if ipString == "127.0.0.1" {
 			return net.ParseIP("10.0.2.2"), nil
+		} else {
```
Comment (medyagh) [inline pkg/minikube/cluster/ip.go:71]: lets add a comment here why this special case

Comment (spowelljr) [reply pkg/minikube/cluster/ip.go:71]: Added a comment to indicate it's for socket_vmnet network

File: pkg/minikube/cluster/ip.go [hunk 2]
```diff
@@ -65,15 +65,13 @@ func HostIP(host *host.Host, clusterName string) (net.IP, error) {
 		if err != nil {
 			return []byte{}, errors.Wrap(err, "Error getting IP address")
 		}
-		// user network case
 		if ipString == "127.0.0.1" {
+			// user network case
```
Comment (medyagh) [inline pkg/minikube/cluster/ip.go:69]: change this to:
// qemu driver: user network case

Comment (spowelljr) [reply pkg/minikube/cluster/ip.go:69]: This is already in `case driver.QEMU:` though
```
	case driver.QEMU2:
		ipString, err := host.Driver.GetIP()
		if err != nil {
			return []byte{}, errors.Wrap(err, "Error getting IP address")
		}
		if ipString == "127.0.0.1" {
			// user network case
			return net.ParseIP("10.0.2.2"), nil
		} else {
			// socket_vmnet network case
			return net.ParseIP("192.168.105.1"), nil
		}
```

Comment (medyagh) [reply pkg/minikube/cluster/ip.go:69]: fair enough !

File: pkg/minikube/cluster/ip.go [hunk 3]
```diff
@@ -65,15 +65,13 @@ func HostIP(host *host.Host, clusterName string) (net.IP, error) {
...
 			return net.ParseIP("10.0.2.2"), nil
+		} else {
+			// socket_vmnet network case
```
[the remark this answers is not in the document]
Comment (spowelljr) [reply pkg/minikube/cluster/ip.go:72]: https://github.com/kubernetes/minikube/pull/15108#discussion_r992800127

**Pull request merged**